File: CompilationTest/BooleanTest/results/BooleanTest.cpp
#include <iostream>

class TestClass {
  public:
    void testMethod() {
        auto a = true;
        auto b = false;
        auto c = a && b;
        auto d = a || b;
        std::cout << std::string("a: ") + (a ? "true" : "false") + ", b: " + (b ? "true" : "false") + ", c: " + (c ? "true" : "false") + ", d: " + (d ? "true" : "false") << std::endl;
    }

  private:
};

int main()
{
    try {
        TestClass c;
        c.testMethod();
    } catch(std::exception& err) {
        std::cout << "Exception: " << err.what() << '\n';
    }
    return 0;
}

File: CompilationTest/BooleanTest/results/BooleanTest.rb
class TestClass
  def test_method()
      a = true
      b = false
      c = a && b
      d = a || b
      puts "a: #{a}, b: #{b}, c: #{c}, d: #{d}"
  end
end

begin
    TestClass.new().test_method()
rescue Exception => err
    puts "Exception: #{err}"
end